File: backend/schemas/alert.py
from datetime import datetime
from pydantic import BaseModel
from typing import Optional, List, Dict

class AlertBase(BaseModel):
    crime_event_id: Optional[int] = None
    title: str
    description: str
    severity: str = "medium"                      # CRITICAL, HIGH, MEDIUM, LOW
    source: str                                   # prediction, network, decision_support, geo
    status: str = "NEW"                           # NEW, ACKNOWLEDGED, IN_PROGRESS, RESOLVED, DISMISSED
    assigned_user_id: Optional[int] = None
    metadata_payload: Optional[str] = None

class AlertCreate(AlertBase):
    pass

class AlertResponse(AlertBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = {
        "from_attributes": True
    }

class AlertStatusUpdate(BaseModel):
    status: str
    assigned_user_id: Optional[int] = None

class AlertSourceStats(BaseModel):
    source: str
    count: int

class AlertSeverityStats(BaseModel):
    severity: str
    count: int

class AlertSummaryResponse(BaseModel):
    total_active: int
    critical_count: int
    resolved_count: int
    today_count: int
    by_source: List[AlertSourceStats]
    by_severity: List[AlertSeverityStats]

File: frontend/types/alert.ts
export interface Alert {
  id: number;
  crime_event_id: number | null;
  title: string;
  description: string;
  severity: "CRITICAL" | "HIGH" | "MEDIUM" | "LOW";
  source: "prediction" | "network" | "decision_support" | "geo";
  status: "NEW" | "ACKNOWLEDGED" | "IN_PROGRESS" | "RESOLVED" | "DISMISSED";
  assigned_user_id: number | null;
  metadata_payload: string | null;
  created_at: string;
  updated_at: string;
}

export interface AlertSourceStats {
  source: string;
  count: number;
}

export interface AlertSeverityStats {
  severity: string;
  count: number;
}

export interface AlertSummary {
  total_active: number;
  critical_count: number;
  resolved_count: number;
  today_count: number;
  by_source: AlertSourceStats[];
  by_severity: AlertSeverityStats[];
}
